dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -0,0 +1,620 @@
+        processed = fn.slice(data, anch, sh, axes=0, device=device)
+        pipe.set_outputs(processed)
+        return pipe
+
+    check_pipeline(generate_data(31, 13, (160, 80, 3), lo=0, hi=255, dtype=np.uint8),
+                   pipeline_fn=pipe)
+
+
+def test_bb_flip():
+    check_pipeline(generate_data(31, 13, (200, 4)), single_op_pipeline, operator_fn=fn.bb_flip)
+
+
+def test_1_hot():
+    check_pipeline(generate_data(31, 13, (200,), lo=0, hi=255, dtype=np.uint8), single_op_pipeline,
+                   operator_fn=fn.one_hot)
+
+
+def test_bbox_paste():
+    def pipe(max_batch_size, input_data, device):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        data = fn.external_source(source=input_data, cycle=False, device=device)
+        paste_posx = fn.uniform(range=(0, 1))
+        paste_posy = fn.uniform(range=(0, 1))
+        paste_ratio = fn.uniform(range=(1, 2))
+        processed = fn.bbox_paste(data, paste_x=paste_posx, paste_y=paste_posy, ratio=paste_ratio)
+        pipe.set_outputs(processed)
+        return pipe
+
+    check_pipeline(generate_data(31, 13, (200, 4)), pipe, eps=.5, devices=['cpu'])
+
+
+def test_coord_flip():
+    def pipe(max_batch_size, input_data, device):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        data = fn.external_source(source=input_data, cycle=False, device=device)
+        processed = fn.coord_flip(data)
+        pipe.set_outputs(processed)
+        return pipe
+
+    check_pipeline(generate_data(31, 13, (200, 2)), pipe)
+
+
+def test_lookup_table():
+    def pipe(max_batch_size, input_data, device):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        data = fn.external_source(source=input_data, cycle=False, device=device)
+        processed = fn.lookup_table(data, keys=[1, 3], values=[10, 50])
+        pipe.set_outputs(processed)
+        return pipe
+
+    check_pipeline(generate_data(31, 13, (313,), lo=0, hi=5, dtype=np.uint8), pipe)
+
+
+def test_reduce():
+    reduce_fns = [
+        fn.reductions.std_dev,
+        fn.reductions.variance
+    ]
+
+    def pipe(max_batch_size, input_data, device, /, reduce_fn):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        data = fn.external_source(source=input_data, cycle=False, device=device)
+        mean = fn.reductions.mean(data)
+        reduced = reduce_fn(data, mean)
+        pipe.set_outputs(reduced)
+        return pipe
+
+    for rf in reduce_fns:
+        check_pipeline(generate_data(31, 13, (313, 131, 2)), pipe, reduce_fn=rf)
+
+
+def test_arithm_ops():
+    def pipe(max_batch_size, input_data, device):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        data = fn.external_source(source=input_data, cycle=False, device=device)
+        data = data * 2
+        data = data + 3
+        data = data - 4
+        data = data / 5
+        data = data // 6
```

Comment:
Can you check unary `-`, ternary `math.clamp`, and throw in a binary op with two DALI tensor arguments, so something like `data + data2`?

Reply:
Done